metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_v2.py

```diff
@@ -453,6 +460,25 @@ def get_workunits(self) -> Iterable[WorkUnit]:
         if self.config.profiling.enabled and len(databases) != 0:
             yield from self.profiler.get_workunits(databases)
 
+        discovered_tables: List[str] = [
+            self.get_dataset_identifier(table.name, schema.name, db.name)
+            for db in databases
+            for schema in db.schemas
+            for table in schema.tables
+        ]
+        discovered_views: List[str] = [
+            self.get_dataset_identifier(table.name, schema.name, db.name)
+            for db in databases
+            for schema in db.schemas
+            for table in schema.views
+        ]
+        discovered_datasets = discovered_tables + discovered_views
+
+        if self.config.include_table_lineage and len(discovered_datasets) != 0:
```

Comment:
done.

Reply:
refractored this bit. Reporting the case where "0 tables and  0 views are found" as an ingestion failure. We would expect at least one table or view to be discovered during ingestion. Otherwise, it is permissions error. No real snowflake ingestion will run without a single table/view.

Reply:
yes